src/command/publish.rs

```diff
@@ -1,15 +1,47 @@
+use self::Access::{Public, Restricted};
 use command::utils::{find_pkg_directory, set_crate_path};
 use error::Error;
 use npm;
 use slog::Logger;
 use std::path::PathBuf;
 use std::result;
+use std::str::FromStr;
 use PBAR;
 
+#[derive(Debug)]
+pub enum Access {
+    Public,
+    Restricted,
+}
+
+impl FromStr for Access {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Error> {
+        return match s {
+      "public" => Ok(Access::Public),
+      "restricted" => Ok(Access::Restricted),
+      "private" => Ok(Access::Restricted),
+      _ => Err(Error::Unsupported { message: format!("{} is not a supported access level. See https://docs.npmjs.com/cli/access for more information on npm package access levels.", s)}),
+    };
+    }
```

Comment:
also really not sure what's up with the formating here, i ran cargo fmt but this is nasty